common/txmgr/resender.go

```diff
@@ -77,7 +77,7 @@ func NewResender[
 	if txConfig.ResendAfterThreshold() == 0 {
 		panic("Resender requires a non-zero threshold")
 	}
-	// todo: add context to evmTxStore
+	// todo: add context to txStore
```

Comment:
Is there a ticket to track this? If so, can we please link it here?

Reply:
Yes, I see the TxStore currently doesn't share the Context with parent components, in case things need to be canceled at shutdown.
It uses a custom pg.QOpt in some places, not all.
We should ideally have the EvmTxStore take in a context at creation time, and use that for all DB calls. That creation time context will be cancleed when the parent component needs to shut down.

@aalu1418 , could you create a task under this EPIC, and link it here:
https://smartcontract-it.atlassian.net/browse/BCI-1501?atlOrigin=eyJpIjoiZTE1MTZmZTQ5ZTcwNGIxM2I0MzI1YjUwMGE4MjMwZWIiLCJwIjoiaiJ9

This task could also be used as an initial ticket for someone to ramp-up.